tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -683,6 +685,272 @@ def test_cpu_wrmsr_restore(microvm_factory, cpu_template):
     test_matrix.run_test(_test_cpu_wrmsr_restore)
 
 
+def dump_cpuid_to_file(dump_fname, ssh_conn):
+    """
+    Read CPUID via SSH and dump it into a file.
+    """
+    _, stdout, stderr = ssh_conn.execute_command("cpuid --one-cpu")
+    assert stderr.read() == ""
+
+    with open(dump_fname, "w", encoding="UTF-8") as file:
+        file.write(stdout.read())
+
+
+def _test_cpu_cpuid_snapshot(context):
+    shared_names = context.custom["shared_names"]
+    root_disk = context.disk.copy(file_name=shared_names["rootfs_fname"])
+    vm_builder = context.custom["builder"]
+    cpu_template = context.custom["cpu_template"]
+
+    vm_instance = vm_builder.build(
+        kernel=context.kernel,
+        disks=[root_disk],
+        ssh_key=context.disk.ssh_key(),
+        config=context.microvm,
+        diff_snapshots=True,
+        cpu_template=cpu_template,
+    )
+
+    vm = vm_instance.vm
+    vm.start()
+
+    ssh_connection = net_tools.SSHConnection(vm.ssh_config)
+
+    # Dump CPUID to a file that will be published to S3 for the 2nd part of the test
+    snapshot_artifacts_dir = os.path.join(
+        shared_names["snapshot_artifacts_root_dir_cpuid"],
+        context.kernel.base_name(),
+        cpu_template if cpu_template else "none",
```

Comment:
Would "no_cpu_template" be more descriptive (though longer)?

Reply:
Not quite see what you mean. Could you expand?

Reply:
When I saw `"none"` I immediately thought it could be a misspelling of Python's `None`, but I think it means `no_cpu_template`. No need to change, more of a nit really.

Reply:
This looks a bit ugly, because the test framework wants None when no template is used, but I need a string to be used as a directory. Not sure how it can be expressed simpler.